### ISSUE TITLE
replace isComplex(x) with isComplex(result)

### ISSUE THREAD

**Issue by axkr** [paulmasson/math#15]
https://github.com/paulmasson/math/blob/5ca4e14ce182c9dcddcacca159baa03e836bc2a6/src/functions/gamma.js#L314

Should the pattern 
```
if ( isComplex(x) ) return result;
```
be replaced with 
```
if ( isComplex(result) ) return result;
```
in the functions `cosIntegral, coshIntegral, sinIntegral, sinhIntegral` ?

**Comment by paulmasson** [comment]
No: `result` is always complex because of the intermediate logarithms of negative numbers. For real input the imaginary part is zero and should be discarded.

**Comment by axkr** [comment]
See this case for negative numbers
- https://www.wolframalpha.com/input/?i=CoshIntegral%5B-4.0%5D

**Comment by paulmasson** [comment]
My bad: `cosIntegral` and `coshIntegral` are both complex for negative real arguments. Fixed in cd9d18e75a314bab9eb61c36b8d5abfe8d8f9d90. Let me know if you find other problematic cases, otherwise closing again.
